otj-metrics-core/src/main/java/com/opentable/metrics/http/HealthController.java

```diff
@@ -13,83 +13,44 @@
-import org.apache.commons.lang3.tuple.Pair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.core.env.ConfigurableEnvironment;
 
 import com.opentable.metrics.health.HealthConfiguration;
-import com.opentable.spring.PropertySourceUtil;
+import com.opentable.metrics.common.CheckController;
 
 @Named
-public class HealthController {
+public class HealthController extends CheckController<Result> {
     private static final Logger LOG = LoggerFactory.getLogger(HealthController.class);
     private static final String CONFIG_PREFIX = "ot.metrics.health.group.";
-    private static final String WARN_PREFIX = "WARN: ";
-
-    private final Map<String, Result> failingChecks = new ConcurrentHashMap<>();
-    private final Map<String, Set<String>> groups = new HashMap<>();
 
     private final HealthCheckRegistry registry;
-    private final ExecutorService executor;
 
+    @Inject
     public HealthController(HealthCheckRegistry registry, @Named(HealthConfiguration.HEALTH_CHECK_POOL_NAME) ExecutorService executor,
             ConfigurableEnvironment env) {
+        super(executor, env, CONFIG_PREFIX);
         this.registry = registry;
-        this.executor = executor;
-        final Properties groupBaseConf = PropertySourceUtil.getProperties(env, CONFIG_PREFIX);
-        groupBaseConf.stringPropertyNames().forEach(group -> {
-            final Set<String> groupItems = Collections.unmodifiableSet(
-                    new HashSet<>(Arrays.asList(groupBaseConf.getProperty(group).split(","))));
-            groups.put(group, groupItems);
-        });
     }
 
-    public Pair<Map<String, Result>, CheckState> runHealthChecks() {
-        final SortedMap<String, Result> checkResults = getCheckResults();
-        final CheckState state = checkResults.values().stream()
-                .map(HealthController::resultToState)
-                .max(CheckState.SEVERITY)
-                .orElse(CheckState.HEALTHY);
-        return Pair.of(checkResults, state);
-    }
-
-    public Pair<Map<String, Result>, CheckState> runHealthChecks(String group) {
-        final Set<String> groupItems = groups.get(group);
-        if (groupItems == null) {
-            return null;
-        }
-        final SortedMap<String, Result> checkResults = getCheckResults();
-        final CheckState state = checkResults.keySet().stream()
-                .filter(groupItems::contains)
-                .map(checkResults::get)
-                .map(HealthController::resultToState)
-                .max(CheckState.SEVERITY)
-                .orElse(CheckState.HEALTHY);
-        final Map<String, Result> toReturn = Maps.filterKeys(checkResults, groupItems::contains);
-        return Pair.of(toReturn, state);
+    @Override
+    protected CheckState resultToState(Result r) {
+        return resToState(r);
     }
 
-    private static CheckState resultToState(Result r) {
+    private static CheckState resToState(Result r) {
```

Comment:
Why not just put this in the protected method?

Reply:
It had to have a static exposure as well as thhe common code.